File: truvornex_android/app/src/main/kotlin/com/truvornex/app/ui/screens/auth/OnboardingScreen.kt
package com.truvornex.app.ui.screens.auth

import androidx.compose.foundation.*
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.rounded.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.truvornex.app.ui.components.TxButton
import com.truvornex.app.ui.theme.*

@Composable
fun OnboardingScreen(
    onCustomer: () -> Unit,
    onProvider: () -> Unit
) {
    var selected by remember { mutableIntStateOf(0) }

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(DarkBg)
            .padding(24.dp)
    ) {
        Column(modifier = Modifier.fillMaxSize()) {
            Spacer(Modifier.weight(1f))

            Text(
                "How will you\nuse Truvornex?",
                style = MaterialTheme.typography.displayLarge.copy(color = DarkPrimary)
            )
            Spacer(Modifier.height(8.dp))
            Text(
                "You can switch between roles anytime.",
                style = MaterialTheme.typography.bodyMedium.copy(color = DarkTextMuted)
            )

            Spacer(Modifier.height(36.dp))

            RoleCard(
                index = 0,
                selected = selected,
                icon = Icons.Rounded.Person,
                title = "I need services",
                subtitle = "Book trusted providers in your neighborhood",
                onClick = { selected = 0 }
            )

            Spacer(Modifier.height(12.dp))

            RoleCard(
                index = 1,
                selected = selected,
                icon = Icons.Rounded.Storefront,
                title = "I offer services",
                subtitle = "Grow your business and manage bookings",
                onClick = { selected = 1 }
            )

            Spacer(Modifier.weight(1f))

            TxButton(
                label = "Get Started",
                onClick = { if (selected == 0) onCustomer() else onProvider() },
                trailingIcon = Icons.Rounded.ArrowForward
            )
            Spacer(Modifier.height(16.dp))
        }
    }
}

@Composable
private fun RoleCard(
    index: Int,
    selected: Int,
    icon: ImageVector,
    title: String,
    subtitle: String,
    onClick: () -> Unit
) {
    val isSelected = selected == index
    val borderColor = if (isSelected) DarkBorderStrong else DarkBorder
    val bgColor = if (isSelected) DarkSurface else DarkSurfaceLow

    Surface(
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick),
        color = bgColor,
        shape = RoundedCornerShape(16.dp),
        border = BorderStroke(if (isSelected) 1.5.dp else 1.dp, borderColor)
    ) {
        Row(
            modifier = Modifier.padding(16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Box(
                modifier = Modifier
                    .size(44.dp)
                    .clip(RoundedCornerShape(12.dp))
                    .background(if (isSelected) DarkPrimary else DarkSurfaceHigh),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    icon,
                    contentDescription = null,
                    tint = if (isSelected) DarkOnPrimary else DarkTextMuted,
                    modifier = Modifier.size(20.dp)
                )
            }
            Spacer(Modifier.width(14.dp))
            Column(modifier = Modifier.weight(1f)) {
                Text(title, style = MaterialTheme.typography.headlineSmall.copy(color = DarkPrimary))
                Spacer(Modifier.height(2.dp))
                Text(subtitle, style = MaterialTheme.typography.bodySmall.copy(color = DarkTextMuted))
            }
            if (isSelected) {
                Icon(Icons.Rounded.CheckCircle, contentDescription = null, tint = DarkSuccess, modifier = Modifier.size(20.dp))
            }
        }
    }
}

File: truvornex_flutter/lib/features/auth/onboarding_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_animate/flutter_animate.dart';
import 'package:go_router/go_router.dart';
import '../../core/theme/app_colors.dart';
import '../../core/theme/app_typography.dart';
import '../../core/theme/app_spacing.dart';
import '../../core/routing/route_names.dart';
import '../shared/widgets/tx_button.dart';

class OnboardingScreen extends StatefulWidget {
  const OnboardingScreen({super.key});

  @override
  State<OnboardingScreen> createState() => _OnboardingScreenState();
}

class _OnboardingScreenState extends State<OnboardingScreen> {
  int _selected = 0; // 0 = customer, 1 = provider

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.darkBg,
      body: SafeArea(
        child: Padding(
          padding: const EdgeInsets.all(AppSpacing.xl),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              const Spacer(),
              Text('How will you\nuse Truvornex?',
                style: AppTypography.display.copyWith(color: AppColors.darkPrimary))
                  .animate().fadeIn(duration: 500.ms)
                  .slideY(begin: 0.1, end: 0, duration: 500.ms, curve: Curves.easeOut),

              const SizedBox(height: 8),
              Text('You can switch between roles anytime.',
                style: AppTypography.body.copyWith(color: AppColors.darkTextMuted))
                  .animate(delay: 100.ms).fadeIn(duration: 400.ms),

              const SizedBox(height: 40),

              _buildRoleCard(
                index: 0,
                icon: Icons.person_outline_rounded,
                title: 'I need services',
                subtitle: 'Book trusted providers in your neighborhood',
              ).animate(delay: 200.ms).fadeIn(duration: 400.ms)
               .slideX(begin: -0.06, end: 0, duration: 400.ms, curve: Curves.easeOut),

              const SizedBox(height: 12),

              _buildRoleCard(
                index: 1,
                icon: Icons.storefront_outlined,
                title: 'I offer services',
                subtitle: 'Grow your business and manage bookings',
              ).animate(delay: 300.ms).fadeIn(duration: 400.ms)
               .slideX(begin: -0.06, end: 0, duration: 400.ms, curve: Curves.easeOut),

              const Spacer(),

              TxButton(
                label: 'Get Started',
                onTap: () => context.goNamed(
                  _selected == 0 ? RouteNames.home : RouteNames.providerHome,
                ),
                icon: Icons.arrow_forward_rounded,
                iconTrailing: true,
              ).animate(delay: 400.ms).fadeIn(duration: 400.ms),

              const SizedBox(height: AppSpacing.base),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildRoleCard({
    required int index,
    required IconData icon,
    required String title,
    required String subtitle,
  }) {
    final isSelected = _selected == index;
    return GestureDetector(
      onTap: () => setState(() => _selected = index),
      child: AnimatedContainer(
        duration: const Duration(milliseconds: 200),
        padding: const EdgeInsets.all(AppSpacing.base),
        decoration: BoxDecoration(
          color: isSelected ? AppColors.darkSurface : AppColors.darkSurfaceLow,
          borderRadius: BorderRadius.circular(AppSpacing.radiusLg),
          border: Border.all(
            color: isSelected ? AppColors.darkBorderStrong : AppColors.darkBorder,
            width: isSelected ? 1.5 : 1,
          ),
        ),
        child: Row(children: [
          Container(
            width: 44,
            height: 44,
            decoration: BoxDecoration(
              color: isSelected ? AppColors.darkPrimary : AppColors.darkSurfaceHigh,
              borderRadius: BorderRadius.circular(12),
            ),
            alignment: Alignment.center,
            child: Icon(icon, size: 20,
              color: isSelected ? AppColors.darkOnPrimary : AppColors.darkTextMuted),
          ),
          const SizedBox(width: AppSpacing.md),
          Expanded(
            child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
              Text(title,
                style: AppTypography.h3.copyWith(color: AppColors.darkPrimary)),
              const SizedBox(height: 3),
              Text(subtitle,
                style: AppTypography.caption.copyWith(color: AppColors.darkTextMuted)),
            ]),
          ),
          if (isSelected)
            const Icon(Icons.check_circle_rounded, size: 20, color: AppColors.darkSuccess),
        ]),
      ),
    );
  }
}
